### PR TITLE
overlap corrections in passive components

### PR DESCRIPTION
**Pull request by amorsch**
corrected
- overlaps between HMP cradle and Frame
- small overlap inside frame
- overlap between RB24 pipe and ZDC (beam pipe)
- overlap between PIPE and COMP
For this a more realistic geometry of the pipe inside COMP has been implemented:
transition round to oval and oval beam.pipe inside the compensator magnet.  
Implementation needs to be reconsidered in case there is a performance penalty.

### CODE REVIEW

**Comment by amorsch** [comment]
I'm not able to reproduce the failed check locally
(testMatBudLUT, o2-test-mch-geometry-regular, o2sim_G4, o2sim_G4_mt, o2sim_hepmc)

**Comment by shahor02** [comment]
@amorsch The testMatBudLUT fails becaus the ``o2-sim -n0`` fails to generate ``o2sim_geometry.root``: for me with your PR it crashes with 
````
...
[12:22:05][INFO] Field in CAVE: 2
[12:22:05][INFO] Setting up CAVE to host ZDC
Info in <TGeoManager::SetTopVolume>: Top volume is cave. Master volume is cave
Info in <TGeoNavigator::BuildCache>: --- Maximum geometry depth set to 100
*** buffer overflow detected ***: o2-sim-serial terminated

(gdb) where
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff047a801 in __GI_abort () at abort.c:79
#2  0x00007ffff04c3897 in __libc_message (action=action@entry=(do_abort | do_backtrace), fmt=fmt@entry=0x7ffff05f0988 "*** %s ***: %s terminated\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff056ecff in __GI___fortify_fail_abort (need_backtrace=need_backtrace@entry=true, msg=msg@entry=0x7ffff05f0905 "buffer overflow detected") at fortify_fail.c:33
#4  0x00007ffff056ed21 in __GI___fortify_fail (msg=msg@entry=0x7ffff05f0905 "buffer overflow detected") at fortify_fail.c:44
#5  0x00007ffff056ca10 in __GI___chk_fail () at chk_fail.c:28
#6  0x00007ffff056bf29 in _IO_str_chk_overflow (fp=<optimized out>, c=<optimized out>) at vsprintf_chk.c:31
#7  0x00007ffff04c8494 in __GI__IO_default_xsputn (f=0x7fffffff7560, data=<optimized out>, n=3) at genops.c:417
#8  0x00007ffff0493ff6 in printf_positional (s=s@entry=0x7fffffff7560, format=format@entry=0x7ffff52f9a09 "voRB242CuOvTransV%d", readonly_format=readonly_format@entry=0, ap=ap@entry=0x7fffffff76a0, ap_savep=ap_savep@entry=0x7fffffff7108, 
    done=done@entry=17, nspecs_done=0, lead_str_end=<optimized out>, work_buffer=<optimized out>, save_errno=<optimized out>, grouping=<optimized out>, thousands_sep=<optimized out>) at vfprintf.c:2022
#9  0x00007ffff0496cba in _IO_vfprintf_internal (s=s@entry=0x7fffffff7560, format=format@entry=0x7ffff52f9a09 "voRB242CuOvTransV%d", ap=ap@entry=0x7fffffff76a0) at vfprintf.c:1688
#10 0x00007ffff056bfcb in ___vsprintf_chk (s=0x7fffffff7880 "voRB242CuOvTransV10", flags=1, slen=20, format=0x7ffff52f9a09 "voRB242CuOvTransV%d", args=args@entry=0x7fffffff76a0) at vsprintf_chk.c:82
#11 0x00007ffff056befa in ___sprintf_chk (s=s@entry=0x7fffffff7880 "voRB242CuOvTransV10", flags=flags@entry=1, slen=slen@entry=20, format=format@entry=0x7ffff52f9a09 "voRB242CuOvTransV%d") at sprintf_chk.c:31
#12 0x00007ffff52d745e in sprintf (__fmt=0x7ffff52f9a09 "voRB242CuOvTransV%d", __s=0x7fffffff7880 "voRB242CuOvTransV10") at /usr/include/x86_64-linux-gnu/bits/stdio2.h:34
#13 o2::passive::Pipe::ConstructGeometry (this=<optimized out>) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/Detectors/Passive/src/Pipe.cxx:1484
#14 0x00007ffff14ee1c4 in FairMCApplication::ConstructGeometry (this=this@entry=0x55555d3d85e0) at /home/shahoian/alice/sw/SOURCES/FairRoot/0/0/base/sim/FairMCApplication.cxx:936
#15 0x00007ffff7bacfa1 in o2::steer::O2MCApplicationBase::ConstructGeometry (this=0x55555d3d85e0) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/Steer/src/O2MCApplication.cxx:90
#16 0x00007fffed5ed948 in TGeant3::Init (this=0x5555609f8a50) at /home/shahoian/alice/sw/SOURCES/GEANT3/21aa79aaa9/0/TGeant3/TGeant3.cxx:6496
#17 0x00007ffff14ec751 in FairMCApplication::InitMC (this=0x55555d3d85e0) at /home/shahoian/alice/sw/SOURCES/FairRoot/0/0/base/sim/FairMCApplication.cxx:433
#18 0x00007ffff7bc33a6 in o2::steer::O2RunSim::Init (this=0x555557b28e00) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/Steer/include/Steer/O2RunSim.h:133
#19 0x0000555555561cde in o2sim_init (asservice=<optimized out>) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/run/../macro/o2sim.C:114
#20 0x00005555555633e1 in o2sim (asservice=false) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/run/../macro/o2sim.C:222
#21 0x000055555555d142 in main (argc=<optimized out>, argv=<optimized out>) at /home/shahoian/alice/sw/SOURCES/O2/v1.2.0/0/run/o2sim.cxx:30
````

**Comment by amorsch** [comment]
Thanks @shahor02 ! Stupid array overflow.  I'll fix it asap.

File: Detectors/Passive/src/FrameStructure.cxx [hunk 1]
```diff
@@ -429,15 +429,19 @@ void FrameStructure::ConstructGeometry()
...
   shB77B->SetName("shB77B");
   TGeoTranslation* trB77A = new TGeoTranslation("trB77A", +283.32, 0., 0.);
   TGeoTranslation* trB77B = new TGeoTranslation("trB77B", -283.32, 0., 0.);
   trB77A->RegisterYourself();
   trB77B->RegisterYourself();
-  TGeoCompositeShape* shB77 = new TGeoCompositeShape("shB77", "shB77A+shB77B:trB77A+shB77B:trB77B");
+  TGeoPgon* shB77C = new TGeoPgon(60., 360., 18, 2);
+  shB77C->SetName("shB77C");
+  shB77C->DefineSection(0, -zE, 422.15, 423.70);
+  shB77C->DefineSection(1, zE, 422.15, 423.70);
```
Comment (sawenzel) [inline Detectors/Passive/src/FrameStructure.cxx:443]: This looks like a tube. We should avoid using complex solids when they can be represented as simple entities for performance reasons.

Comment (amorsch) [reply Detectors/Passive/src/FrameStructure.cxx:443]: The boundary between FRAME and EMC is a polygon. If I change the shape to a TubeSeg
I get either extrusions of elements in FRAME or overlaps with EMC.

Comment (sawenzel) [reply Detectors/Passive/src/FrameStructure.cxx:443]: Ok. From the looks of it this shouldn't make a difference. But let's leave it like this.

**Pull request merged**